lib/cloudformation_rspec/matchers/validate.rb

```diff
@@ -11,12 +12,34 @@ def validate_cf_template(template_body)
     end
     true
   end
+
+  def lint_cf_template(template_body)
+    # Issue a warning if cfn-lint is not installed, but pass the test
+    unless cfn_lint_available
+      warn "Failed to run cfn-lint, do you have it installed and available in $PATH?"
+      return true
+    end
+
+    Tempfile.open(['cfn-lint', '.json']) do |f|
+      f.write(template_body)
+      f.flush
+      _, stderr, status = Open3.capture3('cfn-lint', f.path)
+      if !status
+        @error = stderr
+      end
+      status
+    end
+  end
+
+  def cfn_lint_available
+    !system('cfn-lint', '--version').nil?
```

Comment:
I'm not sure how to do this check in a nice way, but there are two cfn-lint projects:

- https://www.npmjs.com/package/cfn-lint
- https://github.com/awslabs/cfn-python-lint

Stackmaster is only compatible with the python version and  I've seen people get confused by this before.
Is checking that `head -n1 $(which cfn-lint)` contains `python` a terrible idea?

Reply:
The version output looks different, so we could use that? On my machine at least (due to pyenv I assume) which doens't return anything
```
12:11 $ which -a cfn-lint
/usr/local/bin/cfn-lint
✔ ~
12:11 $ cfn-lint --version
cfn-lint 0.9.2
✔ ~
12:11 $ /usr/local/bin/cfn-lint --version
1.9.2
```

Reply:
I wouldn't trust the version... but I think both shbang and version approaches are bad :)

How about checking a specific behavior? Python's `cfn-lint -l` responds with a list of rules and exit code 0. Node's doesn't understand and exits with 1.